Utilities/Exceptions/BranchAndBoundNullParentException.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Utilities.Exceptions
+{
+    /// <summary>
+    ///     Node does not have a parent in Branch and bound knapsack solver.
+    /// </summary>
+    public class BranchAndBoundNullParentException : ApplicationException
```

Comment:
I don't see when it should be used (semantically). Probably it should just be removed.

Reply:
Removed.